crates/polars-plan/src/logical_plan/optimizer/cse/cse_expr.rs

```diff
@@ -115,8 +117,19 @@ mod identifier_impl {
             self.inner.is_some()
         }
 
-        pub fn materialize(&self) -> String {
-            format!("{}{}", CSE_REPLACED, self.inner.unwrap_or(0))
+        pub fn materialize(&self, arena: &Arena<AExpr>) -> String {
+            // This can collide, so we must use strong hashes
+            // The hash of `Series` is very poor, so we traverse it again visit all elements.
+            let last = self.last_node.unwrap();
```

Comment:
This was the issue, the hash of `Series` didn't take all elements into account.

Now we do. To ensure this doesn't get too expensive we set a `SERIES_LIMIT` of 1000 elements.